### ISSUE TITLE
Piecewise() in defintion of equation leads to error in unit checking

### ISSUE THREAD

**Issue by schymans** [schymans/environmental-science-for-sagemath#55]
The unit checking in class Equation leads to an error if we use piecewise definitions, e.g.

```
from sympy import Piecewise
class eq1(Equation):
    """Test"""
    expr = Eq(y, Piecewise(x ** 2, y > 0, 0, >=0))
```

**Comment by jirikuncar** [comment]
```python
from sympy import Piecewise
class eq1(Equation):
     """Test"""
     expr = Eq(y, Piecewise((x ** 2, y > 0), (0, y >= 0)))
```
this should work

**Comment by schymans** [comment]
The failure happens when we use variables with units:
```python
from sympy import Piecewise

class demo_d(Variable):
    """Test variable."""

    unit = meter
    
class demo_d1(Variable):
    """Test variable."""

    unit = meter

    
class t(Variable):
    """Time."""
    unit = second

    
class demo_v(Variable):
    """Test variable."""

    unit = meter/second

class demo_piecewise(Equation):
    """Test piecewise defined equation."""
    
    expr = Eq(demo_d,  Piecewise((demo_d1, t < 0), (demo_v * t, t > 0)))
```
returns

> ValueError: summation of quantities of incompatible dimensions


If we don't use `Equation`, it actually works:
```python
blah = Eq(demo_d,  Piecewise((demo_d1, t < 0), (demo_v * t, t > 0)))
print(blah.subs(t,-1))
print(blah.subs({demo_v: 2, t: 2}))
```
returns 

> Eq(demo_d, demo_d1)
Eq(demo_d, 4)

**Comment by schymans** [comment]
And here is an example showing that the Error happens in `Variable.check_unit()`:
```python
lowt = demo_d1
hight = demo_v * t
pw = Piecewise((lowt, t < 0), (hight, t >=0))
print(Variable.check_unit(demo_d + lowt))
print(Variable.check_unit(demo_d + hight))
try:
    print(Variable.check_unit(demo_d + pw))
except Exception as e1:
    print e1
```
returns

> demo_d + demo_d1
demo_d + demo_v*t
summation of quantities of incompatible dimensions

**Comment by schymans** [comment]
Fixed by #76.
